### PR TITLE
Members identification as uint32

### PR DESCRIPTION
**Pull request by nkuba**
This PR is a follow-up of https://github.com/keep-network/keep-core/pull/2666/files#r732629138

To reduce costs of storing members on-chain we switched from members identification with addresses to unique uint32 identifiers assigned by sortition pool.

Gas usage before this change:
```
|  Contract                ·  Method                                                   ·  Min        ·  Max        ·  Avg        ·  # calls      ·  eur (avg)  │
···························|···························································|·············|·············|·············|···············|··············
|  RandomBeaconStub        ·  submitDkgResult                                          ·     754806  ·    2147836  ·    1921703  ·           74  ·          -  │
```

Gas usage with this change:
```
|  Contract                ·  Method                                                   ·  Min        ·  Max        ·  Avg        ·  # calls      ·  eur (avg)  │
···························|···························································|·············|·············|·············|···············|··············
|  RandomBeaconStub        ·  submitDkgResult                                          ·     867586  ·    1146408  ·    1095996  ·           74  ·          -  │
```

### CODE REVIEW

File: solidity/random-beacon/contracts/libraries/DKG.sol [hunk 1]
```diff
@@ -290,7 +309,8 @@ library DKG {
                 .toEthSignedMessageHash()
                 .recover(current);
             require(
-                members[memberIndex - 1] == recoveredAddress,
+                sortitionPool.getIDOperator(members[memberIndex - 1]) ==
+                    recoveredAddress,
```
[the remark this answers is not in the document]
Comment (pdyraga) [reply solidity/random-beacon/contracts/libraries/DKG.sol:317]: Slither is also complaining about it. I silenced it and added a FIXME in https://github.com/keep-network/keep-core/pull/2677/commits/a2c58970875fe28168378000d57d6f4222175b03.

**Review by pdyraga — APPROVED** [review]
Change in the contracts looks good. We need to optimize the external contract call in `DKG.verify` but it requires changes in the sortition pools repository first. 

We also still need to address outstanding comments from #2676 but we can do it in a separate PR.

Change in tests setup and how we work with `SortitionPoolStub` adds a lot of complexity though and needs to be fixed as soon as possible. It is hard to keep up with all `-1` index operations and where addresses come from. I'll try to do it along with the changes from #2672.

**Comment by pdyraga** [comment]
Merging to unblock other PRs, review comments will be addressed in separate PRs.

**Comment by dimpar** [comment]
wow, avg. gas from 1921703 to 1095996!

File: solidity/random-beacon/contracts/libraries/DKG.sol (hunk 1 above)
[the remark this answers is not in the document]
Comment (pdyraga) [reply solidity/random-beacon/contracts/libraries/DKG.sol:317]: Change on the sortition pool side: https://github.com/keep-network/sortition-pools/pull/126

File: solidity/random-beacon/contracts/libraries/DKG.sol [hunk 2]
```diff
@@ -227,14 +242,18 @@ library DKG {
         uint8[] calldata misbehaved,
         bytes memory signatures,
         uint256[] memory signingMemberIndices,
-        address[] memory members
+        uint32[] calldata members
```
[the remark this answers is not in the document]
Comment (pdyraga) [reply solidity/random-beacon/contracts/libraries/DKG.sol:245]: Done in #2679

**Pull request merged**